src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -7646,6 +7649,77 @@ struct ScatterUpdateComputationConstProp
   };
 };
 
+struct ScatterIndicesAreUnique : public OpRewritePattern<stablehlo::ScatterOp> {
```

Comment:
we also want this for gather too presumably

Reply:
gather only has indices_are_sorted, not the unique indices attr https://github.com/openxla/stablehlo/blob/main/docs/spec.md#inputs-44